src/main/java/frc/robot/controls/Operator.java

```diff
@@ -100,22 +119,27 @@ public Operator(Robot robot)
         climberRetract.whileHeld(new ClimberRetract(robot));
       
         visionAim.whileHeld(new VisionAim(robot));
+        
+        shooterLow.whileHeld(new ShooterSpin(robot, 0.4));
+        shooterMed.whileHeld(new ShooterSpin(robot, 0.46));
+        shooterHigh.whileHeld(new ShooterSpin(robot, 0.48));
+        shooterReverse.whileHeld(new ShooterSpin(robot, -.3));
     }
        // METHODS
 
     // Add methods here which return values for various robot controls by reading the controllers.
 
     /* Authors: Jason Wang & Lucas Jacobs*
-    * Desc:
+    * Desc: NOTNEEDED FOR HUMBER
     * Returns the power given to the shooter by reading the left
     * joystick of the controller*/
 
-    public double getShooterPower(){
+    /*public double getShooterPower(){
         double power = -dualshock.getY();
 
         if (Math.abs(power) < SHOOTER_DEADZONE) power = 0;
 
         return power;
-    }
+    }*/
```

Comment:
If we don't need this anymore you can delete it.  We have multiple manual levels of shooter speeds and we'll also have auto.  If you wanna keep the joystick shooter power thing, then maybe print out the speed on the smart dashboard so that the operator can get an idea of what speed they're at.  If we do this then it might be worthwhile to keep the manual shooter speed in instead of the discrete shooter levels.  So you could have vision print out the theoretical speed and you can see how that compares the the actually speed that you need.  Useful for testing or if vision is being unreliable during the match.

One place where it would be good to always have a hardcoded speed is for the low goal when the robot is right up against the terminal.

Reply:
I don't think we need the joystick shooter speeds anymore. I was planning on replacing the discrete shooter speeds with command groups that get the ball in based off the robot being at specific spots. This would be a backup to the vision. I also noticed at the competiton that the operator would hesitate a lot before shooting to make sure the shooter was spun up. Using command groups would make the robot automatically take a shot as fast as possible.

Reply:
Sound good, I like the idea of having pre-programmed speeds because doing it manually is too difficult to gauge.  Having command groups for shooting at different points also makes sense.  Just make sure that the operator is able to manually trigger those command groups from a button or something.

Reply:
A side note: If we had encoders, we would have been able to measure the speed of the shooter and display it on the dashboard, which would make manual shooting easier for the operator.  But since we are limited in what we know about the robot during the match, having it make most of the timing decisions makes sense.  Usually you know when the shooter is charged based on the noise, but during a match you can't hear anything during the chaos lol.